---
layout: page
title: Teaching
---
---

<p align=center>
Academic Year: 2022/2023

<p>In the second semester, I'll teach the <u>machine (deep) learning module</u> of the course <b>Models and Methods for Computational Biophysics</b> at the Physics Department of Cagliari University (Italy). The remaining modules will be taught by Prof. Attilio V. Vargiu.
A tentative syllabus for the whole course is reported below:
</p>

<p>
- Module I: <u>structure and function of proteins </u> - <br>
</p>
<p>
• Introduction to structure and dynamics of proteins and to their mutual interactions; <br>
• Introduction to protein-protein and protein-ligand binding; <br>
• <u>Practical tutorial</u>: play with the VMD software for protein structure and dynamics visualization. <br>
</p>
<p>
- Module II: <u> machine (deep) learning </u> - <br>
</p>
<p>
• Introduction to machine and deep learning; <br>
• Unsupervised vs supervised learning: similarities and differences;<br>
• Clustering algorithms (hierarchical and partitioning methods); <br>
• The K-means algorithm: theory and examples. <br>
• Defining the optimal number of clusters with the "Elbow" method; <br>
• Introduction to Principal Component Analysis (PCA); <br> 
• Supervised learning: classification vs. regression tasks; <br>
• Building the model: training, validation and test sets; <br>
• The concept of "cost function". K-fold cross validation and overfitting; <br>
• Performance evaluation: the confusion matrix. Sensitivity, specificity, precision and accuracy metrics; <br>
• Support-Vector Machine (SVM) and Random Forest (RF) models; <br>
• Introduction to Neural Networks and Deep Learning;<br>
• <u>Practical tutorial</u> (unsupervised learning): clustering protein conformations with the <br>K-means algorithm; <br>
• <u>Practical tutorial</u> (supervised learning): SVM and RF models to predict ligand-protein complex stability. <br>
</p>
<p>
- Module III: <u> protein-ligand docking </u> - <br>
</p>
<p>
• Introduction to molecular recognition and molecular docking; <br>
• <u>Practical tutorial</u>: run docking calculation with the software "GNINA"; <br>
• <u>Practical tutorial</u>: analyze and process docking results. <br>
</p>
<hr style="height:2px;border-width:0;color:gray;background-color:gray">

<p align=center>


<p>During the first semester, I taught a module named <b>Machine Learning Methods in Computational Biophysics</b> as part of the course of Molecular Modeling of Biological Systems (6 CFU)
taught at Cagliari University (Italy) by Prof. Attilio V. Vargiu.<br>
In particular, the course <a href="https://unica.it/unica/en/ateneo_s07_ss01_sss02.page?mu=Guide/PaginaADErogata.do?ad_er_id=2022*N0*N0*S1*40674*20723&ANNO_\ACCADEMICO=2022&mostra_percorsi=S">(here the link to the syllabus)</a> is thought to give a broad overview of different techniques routinely used in computational biophysics, such as protein sequence alignment, molecular dynamics simulations and molecular docking.
In this context, the machine learning module was thought to give the students an insight into the capabilities of machine learning algorithms when applied to study the biophysical world.
</p>

<p>The syllabus of the <b>Machine Learning module</b>  is reported below. 
</p>

• Introduction to big data, artificial intelligence, machine and deep learning; <br>
• Unsupervised vs supervised learning: similarities and differences;<br>
• The concept of "feature";<br>
• Unsupervised learning: clustering algorithms (hierarchical and partitioning methods); <br>
• The K-means algorithm: theory and examples. Defining the optimal number of clusters with the "Elbow" and "Silhouette" methods. Inizialization techniques (random, Forgy, Kmeans++); <br>
• Introduction to the "Natural language processing" (NLP); <br>
• The Bag-of-Words (BoW) model. K-mers and N-grams; <br>
• Supervised learning: classification vs. regression tasks; <br>
• Building the model: training, validation and test sets; <br>
• The concept of "cost function". K-fold cross validation and overfitting; <br>
• Performance evaluation: the confusion matrix. Sensitivity, specificity, precision and accuracy metrics; <br>
• K-Nearest Neighbors (KNN) and Support-Vector Machines (SVM) models (theory and examples); <br>
• Introduction to python libraries for ML applications (with worked examples); <br>
• <u>Practical tutorial</u> (unsupervised learning): clustering protein conformations with the K-means algorithm; <br>
• <u>Practical tutorial</u> (supervised learning): predicting protein families from DNA sequences of different organisms. <br>




